python/source/state_representation/bind_cartesian_space.cpp

```diff
@@ -11,15 +11,20 @@
 void spatial_state(py::module_& m) {
   py::class_<SpatialState, State> c(m, "SpatialState");
 
-  m.def("swap", py::overload_cast<SpatialState&, SpatialState&>(&state_representation::swap), "Swap the values of the two SpatialStates", "state1"_a, "state2"_a);
+  c.def(py::init<const StateType&>(), "Constructor only specifying the type.", "type"_a);
+  c.def(py::init<const StateType&, const std::string&, const std::string&, const bool&>(), "Constructor with name and reference frame specification.", "type"_a, "name"_a, "reference_frame"_a=std::string("world"), "empty"_a=true);
+  c.def(py::init<const SpatialState&>(), "Copy constructor from another SpatialState.", "state"_a);
 
-  c.def(py::init<const StateType&>(), "Constructor only specifying the type", "type"_a);
-  c.def(py::init<const StateType&, const std::string&, const std::string&, const bool&>(), "Constructor with name and reference frame specification", "type"_a, "name"_a, "reference_frame"_a, "empty"_a=true);
-  c.def(py::init<const SpatialState&>(), "Copy constructor from another SpatialState", "state"_a);
+  c.def("get_reference_frame", &SpatialState::get_reference_frame, "Getter of the reference frame.");
+  c.def("set_reference_frame", &SpatialState::set_reference_frame, "Setter of the reference frame.", "reference_frame"_a);
+  c.def("is_compatible", &SpatialState::is_compatible, "Check if the state is compatible for operations with the state given as argument.", "state"_a);
 
-  c.def("get_reference_frame", &SpatialState::get_reference_frame, "Getter of the reference frame");
-  c.def("set_reference_frame", &SpatialState::set_reference_frame, "Setter of the reference frame");
-  c.def("is_compatible", &SpatialState::is_compatible, "Check if the state is compatible for operations with the state given as argument", "state"_a);
+  c.def("__repr__", [](const SpatialState& state) {
+    std::stringstream buffer;
+    buffer << state;
+    buffer << state;
```

Comment:
```suggestion
    buffer << state;
```

Reply:
Ups, yeah of course haha a relic from debugging